buddy_init/init_actions.py

```diff
@@ -0,0 +1,103 @@
+import os
+import pathlib
+import shutil
+import subprocess
+import venv
+from distutils import dir_util
+
+import fabric
+import paramiko
+from git import Repo
+
+from .init_framework import FatalException
+
+
+def _is_venv():
+    return bool(os.environ.get('VIRTUAL_ENV'))
+
+
+def _call_python(command: str, use_local_venv=True):
+    if use_local_venv:
+        subprocess.check_output(f'venv/bin/python3 -m {command}', shell=True)
+    elif _is_venv():
+        subprocess.check_output(f'python3 -m {command}', shell=True)
+    else:
+        raise FatalException('Installing packages on the system python is not allowed. Aborting')
+
+
+def create_venv():
+    """Create the experiment venv, otherwhise use the already active venv"""
+    venv.EnvBuilder(with_pip=True).create('venv')
+    subprocess.check_output(f'venv/bin/python3 -m pip install --upgrade pip', shell=True)
+
+
+def create_git_repo(base_dir, remote_url):
+    """Setups the git repo with the proper git ignores, it requires a *NEW* the remote ssh address,
+    it looks like: git@github.com:username/example.git"""
+    subprocess.run(f'''(
+        cd {base_dir}
+        touch readme.md
+        git init
+        git add .
+        git commit -m "First commit"
+        git remote add origin {remote_url}
+        git branch -M master
+        git push -f -u origin master
+   )''', capture_output=True, shell=True).stdout.decode()
+
+
+def create_base_structure(use_local_venv=True):
+    """initializes the base example"""
+    base_skeleton_path = f'{os.path.dirname(os.path.realpath(__file__))}/base_skeleton'
+    dir_util.copy_tree(base_skeleton_path, '.')
+    _call_python(command='pip install -r requirements.txt', use_local_venv=use_local_venv)
+
+
+def setup_mila_user(mila_user: str):
+    """One time setup to connect with the Mila servers"""
+    mila_config = f"""
+Host mila1
```

Comment:
maybe separate the string from the core of the function for readability

Reply:
+ too many blank lines in the copied file.

Reply:
Extracted